csrc/scheduler/matmul.cpp

```diff
@@ -718,18 +814,37 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
       {acr, bcr, ab, bb},
       {ParallelType::TIDy, ParallelType::TIDz});
 
+  if (params.has_smem_epilogue) {
+    if (has_epilogue) {
+      // mma_results -> other_tvs -> c_smem -> c
+      scheduler_utils::BoundedDirectionalTransformPropagator::forward(
+          mma_result,
+          -1,
+          ir_utils::producerTvsOf(c_smem),
+          scheduler_utils::BoundedDirectionalTransformPropagator::Options()
+              .propagateParallelType()
+              .propagateToBoundary());
+    }
+    scheduleEpilog(
+        c_smem,
+        mma_result,
+        params,
+        gemm_tile,
+        mma_builder.operand(MmaOptions::Operand::Accumulator).build());
+  }
+
   scheduler_utils::BoundedDirectionalTransformPropagator::forward(
```

Comment:
Are we propagating the transformation of `c_smem` to fusion output? I don't think this makes sense. The purpose for having a smem epilogue is to convert the swizzled memory format in `c_smem` into a more contiguous memory format, so that we can do better gmem storing, for example, vectorize 8 instead of 2.

Reply:
Nice catch! I was extending the usage of propagation when clean up this PR. Didn't realize we want to keep the different schedule of the output tensor!